js/builtins/weakrefs/finalizationregistry-cleanupCallback-throws-onerror-interaction.optional.window.js

```diff
@@ -0,0 +1,69 @@
+// META: script=resources/maybe-garbage-collect.js
+// ├──> maybeGarbageCollectAsync
+// └──> resolveGarbageCollection
+/*---
+esid: sec-finalization-registry-target
+info: |
+  FinalizationRegistry ( cleanupCallback )
+
+  CleanupFinalizationRegistry ( finalizationRegistry [ , callback ] )
+
+  The following steps are performed:
+
+  Assert: finalizationRegistry has [[Cells]] and [[CleanupCallback]] internal slots.
+  If callback is not present or undefined, set callback to finalizationRegistry.[[CleanupCallback]].
+  While finalizationRegistry.[[Cells]] contains a Record cell such that cell.[[WeakRefTarget]] is
+  empty, then an implementation may perform the following steps,
+    Choose any such cell.
+    Remove cell from finalizationRegistry.[[Cells]].
+    Perform ? Call(callback, undefined, « cell.[[HeldValue]] »).
+  Return NormalCompletion(undefined).
+
+  EDITOR'S NOTE
+  When called from HostCleanupFinalizationRegistry, if calling the callback throws an error, this will be caught within the RunJobs algorithm and reported to the host. HTML does not apply the RunJobs algorithm, but will also report the error, which may call window.onerror.
+---*/
+
+let error = new Error('FinalizationRegistryError');
+
+let finalizationRegistry = new FinalizationRegistry(function() {
+  throw error;
+});
+
+setup({ allow_uncaught_exception: true });
+
+assert_implements(
+  typeof FinalizationRegistry.prototype.register === 'function',
+  'FinalizationRegistry.prototype.register is not implemented.'
+);
+
+promise_test((test) => {
```

Comment:
`assert_implements` needs to go inside a subtest, otherwise the test just times out.
```suggestion
promise_test((test) => {
  assert_implements(
    typeof FinalizationRegistry.prototype.register === 'function',
    'FinalizationRegistry.prototype.register is not implemented.'
  );

```